includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -228,6 +228,54 @@ private function filter_attributes( $attributes ) {
 		return $out;
 	}
 
+	/**
+	 * Parse the srcset attribute and deduplicate image candidates that have the same URL and width or pixel density
+	 * defined.
+	 *
+	 * @param DOMElement $node The img element.
+	 */
+	private function validate_srcset( $node ) {
+		$srcset = $node->getAttribute( Attribute::SRCSET );
+
+		if ( empty( $srcset ) ) {
+			return;
+		}
+
+		if ( preg_match_all( '/\s*(?:,\s*)?(?<url>[^,\s]\S*[^,\s])\s*(?<dimension>[\d]+.?[\d]*[wx])?\s*(?:(,)\s*)?/', $srcset, $matches ) ) {
```

Comment:
Otherwise, wouldn't it be possible for a `srcset` to have the value (where no commas are present):

```
https://example.com/image-1.jpg 1024w https://example.com/image-2.jpg 120w
```

But this is supposed to be a validation error:

![image](https://user-images.githubusercontent.com/134745/92986997-85b6fd80-f473-11ea-9a94-30248c9aff7f.png)

Reply:
It appears the JS validator logic solves this problem by omitting the optional `,` from the beginning of the pattern. It only includes the final comma. Then it checks if a comma matched the last time, and if so, it raises an `INVALID_ATTR_VALUE` error.

Reply:
I think I have an alternative solution to the problem -- if the number of parsed URLs does not match the dimensions found, or   there are not enough commas to separate the image candidates, then a validation error is raised:

https://github.com/ampproject/amp-wp/blob/e406fe623d0c046f42b2ce3abf6f0cebf971e961/includes/sanitizers/class-amp-srcset-sanitizer.php#L86-L88